🌐 Add French translation for `docs/fr/docs/index.md` (#4203)
fastapi/fastapi: docs/fr/docs/index.md

**Documentation**: <a href="https://fastapi.tiangolo.com" target="_blank">https://fastapi.tiangolo.com</a>

**Code source**: <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>

---

> FastAPI est un framework Web moderne, rapide (hautes performances) pour la création d'API avec Python 3.6+ basé sur des astuces de type Python standard.

Reviewer: "astuces de type" : je pense que ce serait plus clair pour tout le monde de garder "type hints"

Author: en effet

Ces caractéristiques clés sont:

> * **Rapide**: De trê hautes performances , à égalité avec **NodeJS** et **Go** (grace à Starlette et Pydantic). [l'une des framorks python les plus rapides](#performance).

Reviewer 2: Typo + formattage

```suggestion
* **Rapide**: De très hautes performances, à égalité avec **NodeJS** et **Go** (grace à Starlette et Pydantic). [L'un des framework Python les plus rapides](#performance).
```

> * **Rapide à coder**: augmente ta productivité jusqu'a 200% à 300%. *

Reviewer 2: Traduction plus fidèle à l'original IMO + mise en forme.

```suggestion
* **Rapide à coder**: Augmente la productivité de 200% à 300%. *
```

> * **moins de bugs**: Réduis environ 40 % des erreurs induites par le développeur *

Reviewer 2: Mise en forme

```suggestion
* **Moins de bugs**: Réduis environ 40 % des erreurs induites par le développeur. *
```

Reviewer 3: 40% tout attaché.

> * **Intuitive**: Excellent support pour les éditeurs de code. <abbr title="also known as auto-complete, autocompletion, IntelliSense">Complétiton</abbr> partout. moins de temps pour le débogage.

Reviewer 2: Mise en forme + correction d'une traduction

IMO *Complétion partout* sonne bizarre, pourquoi pas *Complétion intégrée* ou *Auto-complétion intégrée* ?

```suggestion
* **Intuitif**: Excellent support pour les éditeurs de code. <abbr title="aussi connu sous le nom d'auto-complétion, IntelliSense">Complétiton</abbr> partout. Moins de temps passé à débugguer.
```

> * **Facile**: conçu pour être facile à apprendre et à utiliser.moins de temps pour lire la documentation.

Reviewer 2: ```suggestion
* **Facile**: Conçu pour être facile à apprendre et à utiliser. Moins de temps passé à lire la documentation.
```

> * **court**: diminue la duplication de code.plusie Fonctionnalités multiples de chaque déclaration de paramètre. moins de bugs.

Reviewer 2: ```suggestion
* **Concis**: Diminue la duplication de code. De nombreuses fonctionnalités liées à la déclaration de chaque paramètre. Moins de bugs.
```

> * **Robuste**: code prêt pour la production. avec une documentation interactive automatique(Openapi).

Reviewer 2: ```suggestion
* **Robuste**: Code prêt pour la production, avec une documentation interactive automatique.
```

> * **Basé sur des normes**:  basé sur le norme de the open standards for APIs: <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (plus connu sous le nom de Swagger) and <a href="https://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.

Reviewer 2: ```suggestion
* **Basé sur des normes**:  basé, et entièrement compatible, avec les standards ouverts pour les APIs : <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (aussi connu sous le nom de Swagger) et <a href="https://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.
```

> <small>* estimation basée sur des tests sur une équipe de développement interne, construction d'applications de production.</small>

Reviewer 2: ```suggestion
<small>* estimation basée sur des tests d'une équipe de développement interne sur la construction d'applications de production.</small>
```

## Opinions

> "_[...] J'utilise beaucoup **FastAPI** ces derniers jours. [...] j'envisage de le faire utiliser avec mon équipe **ML services chez Microsoft**. Certains d'entre eux s'intègrent avec les produits **Windows** et **Office** "

Reviewer 2: ```suggestion
"_[...] J'utilise beaucoup **FastAPI** ces derniers temps. [...] Je prévois de le faire utiliser par toute mon équipe de **services ML chez Microsoft**. Certains d'entre eux sont intégrés dans le coeur de **Windows** et dans des produits **Office** ".
```

Reviewer 3: ", dont certains sont intégrés"

---

> "_**Netflix** a le plaisir d'annoncer la sortie open source de notre structure d'orchestration de **gestion de crise** : **Dispatch** ! [construit avec **FastAPI**]_"

Reviewer 2: ```suggestion
"_**Netflix** a le plaisir d'annoncer la sortie open source de notre framework d'orchestration de **gestion de crise** : **Dispatch** ! [construit avec **FastAPI**]_"
```

<div style="text-align: right; margin-right: 10%;">Kevin Glisson, Marc Vilanova, Forest Monsen - <strong>Netflix</strong> <a href="https://netflixtechblog.com/introducing-dispatch-da4b8a2a8072" target="_blank"><small>(ref)</small></a></div>

---

> "_Je suis sur la lune enthousiasmé par **FastAPI**. C'est tellement amusant !_"

Reviewer 2: Traduction de *over the moon excited* assez complexe. Je propose *hyper enthousiaste*, le sens est très similaire je trouve. 

```suggestion
"_Je suis hyper enthousiaste par **FastAPI**. C'est tellement amusant !_"
```

Reviewer 3: Je suis enchanté par **FastAPI**.

<div style="text-align: right; margin-right: 10%;">Brian Okken - <strong><a href="https://pythonbytes.fm/episodes/show/123/time-to-right-the-py-wrongs?time_in_sec=855" target="_blank">Python Bytes</a> podcast host</strong> <a href="https://twitter.com/brianokken/status/1112220079972728832" target="_blank"><small>(ref)</small></a></div>

---

> 
> "_Honnêtement, ce que vous avez construit a l'air super solide et poli. À bien des égards, c'est ce que je voulais que **Hug** soit - c'est vraiment inspirant de voir quelqu'un construire ça._"
> 

Reviewer 2: ```suggestion
"_Honnêtement, ce que vous avez construit a l'air super solide et poli. À bien des égards, c'est ce que je voulais que **Hug** soit - c'est vraiment inspirant de voir quelqu'un construire ça._"
```

Reviewer 3: a l'air ... -> paraît très solide et propre.
c'est ce que je voulais ... -> c'est ce que je voulais pour **Hug**
quelqu'un contruire ça -> quelqu'un capable de ça.

plus bas, "créateur de Hug"

<div style="text-align: right; margin-right: 10%;">Timothy Crosley - <strong><a href="https://www.hug.rest/" target="_blank">Hug</a> creator</strong> <a href="https://news.ycombinator.com/item?id=19455465" target="_blank"><small>(ref)</small></a></div>

---

"_Si vous cherchez à apprendre un **framework moderne** pour créer des API REST, consultez **FastAPI** [...] C'est rapide, facile à utiliser et facile à apprendre [...]_"

"_Nous sommes passés à **FastAPI** pour nos **API** [...] Je pense que vous l'aimerez [...]_"

<div style="text-align: right; margin-right: 10%;">Ines Montani - Matthew Honnibal - <strong><a href="https://explosion.ai" target="_blank">Explosion AI</a> foundateur - <a href="https://spacy.io" target="_blank">spaCy</a> créateurs</strong> <a href="https://twitter.com/_inesmontani/status/1144173225322143744" target="_blank"><small>(ref)</small></a> - <a href="https://twitter.com/honnibal/status/1144031421859655680" target="_blank"><small>(ref)</small></a></div>

---

> ## **Typer**, le FastAPI des CLI(command line integration)

Reviewer 2: IMO il n'y a pas besoin de détailler la signification de *CLI*, autrement il faudrait aussi détailler celle de *API*.

```suggestion
## **Typer**, le FastAPI des CLI
```

<a href="https://typer.tiangolo.com" target="_blank"><img src="https://typer.tiangolo.com/img/logo-margin/logo-margin-vector.svg" style="width: 20%;"></a>

> Si vous créez une  <abbr title="Command Line Interface"> CLI </abbr> application à utiliser dans le terminal au lieu d'une API Web, consultez <a href="https://typer.tiangolo.com/" class="external-link" target="_blank">**Typer**</a>.

Reviewer 2: ```suggestion
Si vous créez une  <abbr title="Command Line Interface"> CLI </abbr> à utiliser dans le terminal au lieu d'une API Web, consultez <a href="https://typer.tiangolo.com/" class="external-link" target="_blank">**Typer**</a>.
```

> **Typer** est le petit frère de FastAPI. Et il est destiné à être le **FastAPI des CLI*

Reviewer 2: ```suggestion
**Typer** est le petit frère de FastAPI et est destiné à être le **FastAPI des CLI* ⌨️ 🚀
```

> prérequis

Reviewer 2: ```suggestion
## Prérequis
```

Python 3.6+

FastAPI repose sur les épaules de géants :

* <a href="https://www.starlette.io/" class="external-link" target="_blank">Starlette</a> pour la partie web.

> * <a href="https://pydantic-docs.helpmanual.io/" class="external-link" target="_blank">Pydantic</a> pour la partie donnée.

Reviewer 2: ```suggestion
* <a href="https://pydantic-docs.helpmanual.io/" class="external-link" target="_blank">Pydantic</a> pour la partie données.
```

> 

Reviewer 3: "We adopted the FastAPI library"... à traduire

> "_Nous avons adopté la librairie **FastAPI** pour créer un serveur **REST** pouvant être requêté pour obtenir des **prédictions** [pour Ludwig]_"

<div style="text-align: right; margin-right: 10%;">Kevin Glisson, Marc Vilanova, Forest Monsen - <strong>Netflix</strong> <a href="https://netflixtechblog.com/introducing-dispatch-da4b8a2a8072" target="_blank"><small>(ref)</small></a></div>

---
"_Je suis sur la lune enthousiasmé par **FastAPI**. C'est tellement amusant !_"

<div style="text-align: right; margin-right: 10%;">Brian Okken - <strong><a href="https://pythonbytes.fm/episodes/show/123/time-to-right-the-py-wrongs?time_in_sec=855" target="_blank">Python Bytes</a> podcast host</strong> <a href="https://twitter.com/brianokken/status/1112220079972728832" target="_blank"><small>(ref)</small></a></div>

---

"_Honnêtement, ce que vous avez construit a l'air super solide et poli. À bien des égards, c'est ce que je voulais que **Hug** soit - c'est vraiment inspirant de voir quelqu'un construire ça._"

<div style="text-align: right; margin-right: 10%;">Timothy Crosley - <strong><a href="https://www.hug.rest/" target="_blank">Hug</a> creator</strong> <a href="https://news.ycombinator.com/item?id=19455465" target="_blank"><small>(ref)</small></a></div>

---

> "_Si vous cherchez à apprendre un **framework moderne** pour créer des API REST, consultez **FastAPI** [...] C'est rapide, facile à utiliser et facile à apprendre [...]_"

Reviewer 3: consultez -> regardez

"_Nous sommes passés à **FastAPI** pour nos **API** [...] Je pense que vous l'aimerez [...]_"

> <div style="text-align: right; margin-right: 10%;">Ines Montani - Matthew Honnibal - <strong><a href="https://explosion.ai" target="_blank">Explosion AI</a> foundateur - <a href="https://spacy.io" target="_blank">spaCy</a> créateurs</strong> <a href="https://twitter.com/_inesmontani/status/1144173225322143744" target="_blank"><small>(ref)</small></a> - <a href="https://twitter.com/honnibal/status/1144031421859655680" target="_blank"><small>(ref)</small></a></div>

Reviewer 3: ```suggestion
<div style="text-align: right; margin-right: 10%;">Ines Montani - Matthew Honnibal - fondateurs de <strong><a href="https://explosion.ai" target="_blank">Explosion AI</a> - créateurs de <a href="https://spacy.io" target="_blank">spaCy</a></strong> <a href="https://twitter.com/_inesmontani/status/1144173225322143744" target="_blank"><small>(ref)</small></a> - <a href="https://twitter.com/honnibal/status/1144031421859655680" target="_blank"><small>(ref)</small></a></div>
```

---

## **Typer**, le FastAPI des CLI(command line integration)

<a href="https://typer.tiangolo.com" target="_blank"><img src="https://typer.tiangolo.com/img/logo-margin/logo-margin-vector.svg" style="width: 20%;"></a>

> Si vous créez une  <abbr title="Command Line Interface"> CLI </abbr> application à utiliser dans le terminal au lieu d'une API Web, consultez <a href="https://typer.tiangolo.com/" class="external-link" target="_blank">**Typer**</a>.

Reviewer 3: Si vous voulez créer une CLI utilisable dans un terminal, .... regardez ...

</div>

> Vous aurez également besoin d'un serveur ASGI, pour la production telle que <a href="https://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> ou <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.

Reviewer 2: ```suggestion
Vous aurez également besoin d'un serveur ASGI, pour la production tel que <a href="https://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> ou <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.
```

Reviewer 3: pour la production, comme ...

</div>

## Exemple

> ### Crée le

Reviewer 2: ```suggestion
### Créez
```

> * crée un fichier `main.py` avec:

Reviewer 2: ```suggestion
* Créez un fichier `main.py` avec :
```

```

<details markdown="1">

> <summary>Ou utilise <code>async def</code>...</summary>

Reviewer 2: ```suggestion
<summary>Ou bien utilisez <code>async def</code>...</summary>
```

> si ton code utilise `async` / `await`, utilise `async def`:

Reviewer 2: ```suggestion
Si votre code utilise `async` / `await`, utilisez `async def`:
```

**Note**:

> Si vous ne le savez pas, consultez la section _"in a hurry?"_
>  à propos de <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` et `await` dans les docs</a>.

Reviewer 2: ```suggestion
Si vous n'y êtes pas familiers, consultez la section _"in a hurry?"_ à propos de <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` et `await` dans les docs</a>.
```

</details>

> ### Démarre le

Reviewer 2: ```suggestion
### Lancez
```

> Démarre le server avec:

Reviewer 2: ```suggestion
Lancez le serveur avec :
```

</div>

<details markdown="1">

> <summary>à propos de la commande <code>uvicorn main:app --reload</code>...</summary>

Reviewer 2: ```suggestion
<summary>À propos de la commande <code>uvicorn main:app --reload</code>...</summary>
```

> la commande `uvicorn main:app` fait référence à:

Reviewer 2: ```suggestion
La commande `uvicorn main:app` fait référence à :
```

> * `main`: le fichier `main.py` (the Python "module").

Reviewer 2: ```suggestion
* `main`: le fichier `main.py` (le "module" Python).
```

> * `app`: l'objet crée dans `main.py` avec la ligne `app = FastAPI()`.

Reviewer 2: ```suggestion
* `app`: l'objet créé dans `main.py` avec la ligne `app = FastAPI()`.
```

> * `--reload`: faire redémarrer le serveur après les changements de code. Ne le faites que pour le développement.

Reviewer 2: ```suggestion
* `--reload`: fait redémarrer le serveur après des changements de code. À n'utiliser que pour le développement.
```

</details>

### Check it

Ouvrez votre navigateur sur <a href="http://127.0.0.1:8000/items/5?q=somequery" class="external-link" target="_blank">http://127.0.0.1:8000/items/5?q=somequery</a>.

Vous verrez la réponse JSON comme :
```JSON
{"item_id": 5, "q": "somequery"}
```

Vous avez déjà créé une API qui :

> * Reçoit les requêtes HTTP dans les _paths_ `/` et `/items/{item_id}`.

Reviewer 2: ```suggestion
* Reçoit les requêtes HTTP pour les _chemins_ `/` et `/items/{item_id}`.
```

> * Les deux _chemins_ acceptent les <em>operations</em> (également connu sous le nom de méthode HTTP).

Reviewer 2: ```suggestion
* Les deux _chemins_ acceptent des <em>operations</em> `GET` (également connu sous le nom de méthode HTTP).
```

Reviewer 3: opérations, avec un accent aigu

> * le _chemin_ `/items/{item_id}` à un paramétre `item_id` qui doit être un `int`.

Reviewer 2: ```suggestion
* le _chemin_ `/items/{item_id}` à un paramètre `item_id` qui doit être un `int`.
```

> * le _chemin_ `/items/{item_id}` a un paramétre optionnel `q`.

Reviewer 2: ```suggestion
* le _chemin_ `/items/{item_id}` a un paramètre optionnel `q`.
```

</div>

<details markdown="1">
<summary>à propos de la commande <code>uvicorn main:app --reload</code>...</summary>

la commande `uvicorn main:app` fait référence à:

* `main`: le fichier `main.py` (the Python "module").
* `app`: l'objet crée dans `main.py` avec la ligne `app = FastAPI()`.
* `--reload`: faire redémarrer le serveur après les changements de code. Ne le faites que pour le développement.

</details>

### Check it

Ouvrez votre navigateur sur <a href="http://127.0.0.1:8000/items/5?q=somequery" class="external-link" target="_blank">http://127.0.0.1:8000/items/5?q=somequery</a>.

> Vous verrez la réponse JSON comme :

Reviewer 3: ```suggestion
Vous obtenez cette réponse JSON :
```

```JSON
{"item_id": 5, "q": "somequery"}
```

> Vous avez déjà créé une API qui :

Reviewer 3: ```suggestion
Vous venez de créer une API qui :
```

* Reçoit les requêtes HTTP dans les _paths_ `/` et `/items/{item_id}`.
* Les deux _chemins_ acceptent les <em>operations</em> (également connu sous le nom de méthode HTTP).
* le _chemin_ `/items/{item_id}` à un paramétre `item_id` qui doit être un `int`.
* le _chemin_ `/items/{item_id}` a un paramétre optionnel `q`.

> ### Interactive API docs

Reviewer 3: ```suggestion
### Documentation API interactive
```

### Alternative API docs

> Et maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

Reviewer 2: ```suggestion
Maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
```

> Vous verrez la documentation automatique alternative (provided by <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>):

Reviewer 2: ```suggestion
Vous verrez la documentation automatique alternative (fournie par <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>) :
```

Reviewer 3: voyez au présent

> ### Alternative API docs

Reviewer 3: ```suggestion
### Documentation API alternative
```

> Et maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

Reviewer 3: visitez

Vous verrez la documentation automatique alternative (provided by <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>):

![ReDoc](https://fastapi.tiangolo.com/img/index/index-02-redoc-simple.png)

> ## Exemple de mise à niveau

Reviewer 3: ```suggestion
## Exemple plus poussé
```

Modifiez maintenant le fichier `main.py` pour recevoir un corps d'une requête `PUT`.

> Déclarez le corps en utilisant les types Python standards, grâce à Pydantic.

Reviewer 3: ```suggestion
Déclarez ce corps en utilisant les types Python standards, grâce à Pydantic.
```

    return {"item_name": item.name, "item_id": item_id}
```

Le serveur devrait se recharger automatiquement (parce que vous avez ajouté `--reload` à la commande `uvicorn` ci-dessus).

### Mise à niveau interactive de la documentation de l'API

> Et maintenant, allez à <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>.

Reviewer 2: ```suggestion
Maintenant, allez à <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>.
```

* La documentation interactive de l'API sera automatiquement mise à jour, y compris le nouveau corps :

![Swagger UI](https://fastapi.tiangolo.com/img/index/index-03-swagger-02.png)

* Cliquez sur le bouton "Try it out", il vous permet de renseigner les paramètres et d'interagir directement avec l'API :

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-04-swagger-03.png)

> * Cliquez ensuite sur le bouton « Exécuter », l'interface utilisateur communiquera avec votre API, enverra les paramètres, obtiendra les résultats et les affichera à l'écran :

Reviewer 2: ```suggestion
* Cliquez ensuite sur le bouton « Execute », l'interface utilisateur communiquera avec votre API, enverra les paramètres, obtiendra les résultats et les affichera à l'écran :
```

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-05-swagger-04.png)

### Mise à niveau de la documentation API alternative

> Et maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

Reviewer 2: ```suggestion
Maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
```

* La documentation alternative reflétera également le nouveau paramètre de requête et le nouveau corps :

![ReDoc](https://fastapi.tiangolo.com/img/index/index-06-redoc-02.png)

### Recap

En résumé, vous déclarez **une fois** les types de paramètres, corps, etc. en tant que paramètres de fonction.

Vous faites cela avec les types Python modernes standard.

Vous n'avez pas besoin d'apprendre une nouvelle syntaxe, les méthodes ou les classes d'une bibliothèque spécifique, etc.

> Juste standard **Python 3.6+**.

Reviewer 2: ```suggestion
Juste du **Python 3.6+ standard**.
```

> par exemple, pour un `int`:

Reviewer 2: ```suggestion
Par exemple, pour un `int`:
```

```Python
item_id: int
```

ou pour un modèle 'Item' plus complexe :

```Python
item: Item
```

...et avec cette seule déclaration vous obtenez 

> * Assistance à l'éditeur, notamment :
>     * auto-complétion.
>     * verification des types.
> * Validation des données :
>     * Erreurs automatiques et claires lorsque les données ne sont pas valides.
>     * Validation même pour les objets JSON profondément imbriqués.
> * <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données d'entrée : provenant du réseau vers les données et types Python. Lecture de :
>     *  JSON.
>     * Paramètres de chemin.
>     * Paramètres de requête.
>     * cookies.
>     * headers.
>     * Formulaires.
>     * Fichiers.
> 
> * <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données de sortie : conversion des données et types Python en données réseau (au format JSON) :
>     * Convertir les types Python (`str`, `int`, `float`, `bool`, `list`, etc.).
>     * objets `datetime`.
>     * Objets `UUID`.
>     * Modèles de base de données.
>     * ...et beaucoup plus.
> * Documentation API interactive automatique, comprenant 2 interfaces utilisateur alternatives :
>     * Interface utilisateur Swagger.
>     * Redoc.

Reviewer 2: ```suggestion
* Assistance à l'éditeur, notamment :
    * Auto-complétion.
    * Vérification des types.
* Validation des données :
    * Erreurs automatiques et claires lorsque les données ne sont pas valides.
    * Validation même pour les objets JSON profondément imbriqués.
* <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données d'entrée : provenant du réseau vers les données et types Python. Lecture de :
    * JSON.
    * Paramètres de chemin.
    * Paramètres de requête.
    * Cookies.
    * En-têtes.
    * Formulaires.
    * Fichiers.

* <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données de sortie : conversion des données et types Python en données réseau (au format JSON) :
    * Conversion des types Python (`str`, `int`, `float`, `bool`, `list`, etc.).
    * Objets `datetime`.
    * Objets `UUID`.
    * Modèles de base de données.
    * ...et beaucoup plus.
* Documentation API interactive automatique, comprenant 2 interfaces utilisateur alternatives :
    * Interface utilisateur Swagger.
    * Redoc.
```

---

Pour revenir à l'exemple de code précédent, **FastAPI** :

> * Validez qu'il y a un `item_id` dans le chemin pour les requêtes `GET` et `PUT`.
> * Validez que le `item_id` est de type `int` pour les requêtes `GET` et `PUT`.
>     * Si ce n'est pas le cas, le client verra une erreur utile et claire.
> * Vérifiez s'il existe un paramètre de requête facultatif nommé `q` (comme dans `http://127.0.0.1:8000/items/foo?q=somequery`) pour les requêtes `GET`.
>     * Comme le paramètre `q` est déclaré avec `= None`, il est facultatif.
>     * Sans le `None` il serait nécessaire (comme c'est le cas dans le cas avec `PUT`).
> * Pour les requêtes `PUT` vers `/items/{item_id}`, lisez le corps en JSON :
> * Vérifiez qu'il a un attribut obligatoire `name` qui devrait être un `str`.
>     * Vérifiez qu'il a un attribut obligatoire « prix » qui doit être un « flotteur ».
>     * Vérifiez qu'il a un attribut facultatif `is_offer`, qui devrait être un `bool`, s'il est présent.
>     * Tout cela fonctionnerait également pour les objets JSON profondément imbriqués.
> * Convertir de et vers JSON automatiquement.
> * Documentez tout avec OpenAPI, qui peut être utilisé par :
>     * Systèmes de documentation interactifs.
>     * Systèmes de génération automatique de code client, pour de nombreuses langues.
> * Fournir directement 2 interfaces web de documentation interactive.

Reviewer 2: ```suggestion
* Valide qu'il y a un `item_id` dans le chemin pour les requêtes `GET` et `PUT`.
* Valide que le `item_id` est de type `int` pour les requêtes `GET` et `PUT`.
    * Si ce n'est pas le cas, le client verra une erreur utile et claire.
* Vérifie s'il existe un paramètre de requête facultatif nommé `q` (comme dans `http://127.0.0.1:8000/items/foo?q=somequery`) pour les requêtes `GET`.
    * Comme le paramètre `q` est déclaré avec `= None`, il est facultatif.
    * Sans le `None` il serait nécessaire (comme c'est le cas dans le cas avec `PUT`).
* Pour les requêtes `PUT` vers `/items/{item_id}`, lit le corps en JSON :
    * Vérifie qu'il a un attribut obligatoire `name` qui devrait être un `str`.
    * Vérifie qu'il a un attribut obligatoire « prix » qui doit être un « flotteur ».
    * Vérifie qu'il a un attribut facultatif `is_offer`, qui devrait être un `bool`, s'il est présent.
    * Tout cela fonctionnerait également pour les objets JSON profondément imbriqués.
* Convertit de et vers JSON automatiquement.
* Documente tout avec OpenAPI, qui peut être utilisé par :
    * Systèmes de documentation interactifs.
    * Systèmes de génération automatique de code client, pour de nombreuses langues.
* Fournit directement 2 interfaces web de documentation interactive.
```

    return {"item_name": item.name, "item_id": item_id}
```

> Le serveur devrait se recharger automatiquement (parce que vous avez ajouté `--reload` à la commande `uvicorn` ci-dessus).

Reviewer 3: ```suggestion
Le serveur se recharge normalement automatiquement (car vous avez pensé à `--reload` dans la commande `uvicorn` ci-dessus).
```

> ### Mise à niveau interactive de la documentation de l'API

Reviewer 3: ```suggestion
### Plus loin avec la documentation API interactive
```

Et maintenant, allez à <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>.

* La documentation interactive de l'API sera automatiquement mise à jour, y compris le nouveau corps :

![Swagger UI](https://fastapi.tiangolo.com/img/index/index-03-swagger-02.png)

* Cliquez sur le bouton "Try it out", il vous permet de renseigner les paramètres et d'interagir directement avec l'API :

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-04-swagger-03.png)

* Cliquez ensuite sur le bouton « Exécuter », l'interface utilisateur communiquera avec votre API, enverra les paramètres, obtiendra les résultats et les affichera à l'écran :

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-05-swagger-04.png)

> ### Mise à niveau de la documentation API alternative

Reviewer 3: ```suggestion
### Plus loin avec la documentation API alternative
```

Et maintenant, allez à <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

* La documentation alternative reflétera également le nouveau paramètre de requête et le nouveau corps :

![ReDoc](https://fastapi.tiangolo.com/img/index/index-06-redoc-02.png)

> ### Recap

Reviewer 3: ```suggestion
### En résumé
```

En résumé, vous déclarez **une fois** les types de paramètres, corps, etc. en tant que paramètres de fonction.

Vous faites cela avec les types Python modernes standard.

Vous n'avez pas besoin d'apprendre une nouvelle syntaxe, les méthodes ou les classes d'une bibliothèque spécifique, etc.

Juste standard **Python 3.6+**.

par exemple, pour un `int`:

```Python
item_id: int
```

ou pour un modèle 'Item' plus complexe :

```Python
item: Item
```

...et avec cette seule déclaration vous obtenez 

> * Assistance à l'éditeur, notamment :

Reviewer 3: ```suggestion
* une assistance dans l'éditeur, notamment :
```

>     * auto-complétion.

Reviewer 3: ```suggestion
    * l'auto-complétion.
```

>     * verification des types.

Reviewer 3: ```suggestion
    * la vérification des types.
```

> * Validation des données :

Reviewer 3: ```suggestion
* la validation des données :
```

>     * Erreurs automatiques et claires lorsque les données ne sont pas valides.

Reviewer 3: ```suggestion
    * des erreurs automatiques et claires lorsque les données ne sont pas valides.
```

>     * Validation même pour les objets JSON profondément imbriqués.

Reviewer 3: ```suggestion
    * une validation même pour les objets JSON profondément imbriqués.
```

> * <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données d'entrée : provenant du réseau vers les données et types Python. Lecture de :

Reviewer 3: ```suggestion
* <abbr title="also known as: serialization, parsing, marshalling">une conversion</abbr> des données d'entrée : venant du réseau et allant vers les données et types de Python, permettant de lire :
```

>     *  JSON.

Reviewer 3: ```suggestion
    * le JSON.
```

>     * Paramètres de chemin.

Reviewer 3: ```suggestion
    * les paramètres du path.
```

>     * Paramètres de requête.

Reviewer 3: ```suggestion
    * les paramètres de la requête.
```

>     * cookies.

Reviewer 3: ```suggestion
    * les cookies.
```

>     * headers.

Reviewer 3: ```suggestion
    * les en-têtes.
```

>     * Formulaires.

Reviewer 3: ```suggestion
    * les formulaires.
```

>     * Fichiers.

Reviewer 3: ```suggestion
    * les fichiers.
```

> * <abbr title="also known as: serialization, parsing, marshalling">Conversion</abbr> des données de sortie : conversion des données et types Python en données réseau (au format JSON) :

Reviewer 3: ```suggestion
* <abbr title="also known as: serialization, parsing, marshalling">la conversion</abbr> des données de sortie : conversion des données et types de Python en données réseau (au format JSON), permettant de convertir :
```

>     * Convertir les types Python (`str`, `int`, `float`, `bool`, `list`, etc.).

Reviewer 3: ```suggestion
    * les types Python (`str`, `int`, `float`, `bool`, `list`, etc.).
```

>     * objets `datetime`.

Reviewer 3: ```suggestion
    * les objets `datetime`.
```

>     * Objets `UUID`.

Reviewer 3: ```suggestion
    * les objets `UUID`.
```

>     * Modèles de base de données.

Reviewer 3: ```suggestion
    * les modèles de base de données.
```

>     * ...et beaucoup plus.

Reviewer 3: ```suggestion
    * ...et plus encore.
```

> * Documentation API interactive automatique, comprenant 2 interfaces utilisateur alternatives :

Reviewer 3: ```suggestion
* la documentation API interactive automatique, avec 2 interfaces utilisateur au choix :
```

>     * Interface utilisateur Swagger.

Reviewer 3: ```suggestion
    * l'interface utilisateur Swagger.
```

    * Redoc.

---

> Pour revenir à l'exemple de code précédent, **FastAPI** :

Reviewer 3: ```suggestion
Pour revenir à l'exemple de code précédent, **FastAPI** permet de :
```

> * Validez qu'il y a un `item_id` dans le chemin pour les requêtes `GET` et `PUT`.

Reviewer 3: ```suggestion
* validez qu'il y a un `item_id` dans le chemin pour les requêtes `GET` et `PUT`.
```

> * Validez que le `item_id` est de type `int` pour les requêtes `GET` et `PUT`.

Reviewer 3: ```suggestion
* Valider que le `item_id` est de type `int` pour les requêtes `GET` et `PUT`.
```

>     * Si ce n'est pas le cas, le client verra une erreur utile et claire.

Reviewer 3: ```suggestion
    * Si ce n'est pas le cas, le client voit une erreur utile et claire.
```

> * Vérifiez s'il existe un paramètre de requête facultatif nommé `q` (comme dans `http://127.0.0.1:8000/items/foo?q=somequery`) pour les requêtes `GET`.

Reviewer 3: ```suggestion
* Vérifier qu'il existe un paramètre de requête facultatif nommé `q` (comme dans `http://127.0.0.1:8000/items/foo?q=somequery`) pour les requêtes `GET`.
```

>     * Comme le paramètre `q` est déclaré avec `= None`, il est facultatif.

Reviewer 3: ```suggestion
    * puisque le paramètre `q` est déclaré avec `= None`, il est facultatif.
```

>     * Sans le `None` il serait nécessaire (comme c'est le cas dans le cas avec `PUT`).

Reviewer 3: ```suggestion
    * sans le `None` il serait obligatoire (comme c'est le cas dans le cas avec `PUT`).
```

> * Pour les requêtes `PUT` vers `/items/{item_id}`, lisez le corps en JSON :

Reviewer 3: ```suggestion
* Pour les requêtes `PUT` vers `/items/{item_id}`, de lire le corps en JSON :
```

> * Vérifiez qu'il a un attribut obligatoire `name` qui devrait être un `str`.

Reviewer 3: ```suggestion
* vérifier qu'il a un attribut obligatoire `name` qui devrait être un `str`.
```

>     * Vérifiez qu'il a un attribut obligatoire « prix » qui doit être un « flotteur ».

Reviewer 3: ```suggestion
    * vérifier qu'il a un attribut obligatoire « prix » qui doit être un « flotteur ».
```

>     * Vérifiez qu'il a un attribut facultatif `is_offer`, qui devrait être un `bool`, s'il est présent.

Reviewer 3: ```suggestion
    * vérifier qu'il a un attribut facultatif `is_offer`, qui devrait être un `bool`, s'il est présent.
```

        ... "item_price": item.price ...
```

...et voyez comment votre éditeur complétera automatiquement les attributs et connaîtra leurs types :

![editor support](https://fastapi.tiangolo.com/img/vscode-completion.png)
Pour un exemple plus complet incluant plus de fonctionnalités, voir le <a href="https://fastapi.tiangolo.com/tutorial/">Tutoriel - User Guide</a>.

> **Spoiler alert**: le tutoriel - user guide inclu:

Reviewer 2: ```suggestion
**Spoiler alert**: le tutoriel - user guide inclut:
```

* Déclaration de **paramètres** provenant d'autres endroits différents comme : **en-têtes**, **cookies**, **champs de formulaire** et **fichiers**.

> * Comment définir les **contraintes de validation** comme `maximum_length` ou `regex`.

Reviewer 2: ```suggestion
* L'utilisation de **contraintes de validation** comme `maximum_length` ou `regex`.
```

> * Un très puissant et facile à utiliser **<abbr title="also known as components, resources, providers, services, injectables">Systéme d'injection de dépendance </abbr>** .

Reviewer 2: Pas certain qu'il faille traduire *provider*.

```suggestion
* Un **<abbr title="aussi connu sous le nom de composants, ressources, fournisseurs, services, injectables">systéme d'injection de dépendance </abbr>** très puissant et facile à utiliser .
```

> * Sécurité et authentification, y compris la prise en charge de **OAuth2** avec **jetons JWT** et **HTTP Basic** auth.

Reviewer 2: ```suggestion
* Sécurité et authentification, y compris la prise en charge de **OAuth2** avec **jetons JWT** et l'authentification **HTTP Basic**.
```

* Des techniques plus avancées (mais tout aussi simples) pour déclarer des **modèles JSON profondément imbriqués** (grâce à Pydantic).
* De nombreuses fonctionnalités supplémentaires (grâce à Starlette) comme:
    * **WebSockets**
    * **GraphQL**
    * tests extrêmement faciles basés sur `requests` et `pytest`
    * **CORS**
    * **Cookie Sessions**
    * ...et plus encore.

## Performance

Les benchmarks TechEmpower indépendants montrent les applications **FastAPI** s'exécutant sous Uvicorn comme <a href="https://www.techempower.com/benchmarks/#section=test&runid=7464e520-0dc2-473d-bd34-dbdfd7e85911&hw=ph&test=query&l=zijzen-7" class="external-link" target="_blank"> l'un des frameworks Python les plus rapides disponibles </a>, uniquement en dessous de Starlette et Uvicorn eux-mêmes (utilisés en interne par FastAPI). (*)

Pour en savoir plus, consultez la section <a href="https://fastapi.tiangolo.com/benchmarks/" class="internal-link" target="_blank">Benchmarks</a>.

## Dépendances facultatives

Utilisé par Pydantic:

> * <a href="https://github.com/esnme/ultrajson" target="_blank"><code>ujson</code></a> - pour un JSON plus rapide <abbr title="converting the string that comes from an HTTP request into Python data">"parsing"</abbr>.

Reviewer 2: ```suggestion
* <a href="https://github.com/esnme/ultrajson" target="_blank"><code>ujson</code></a> - pour un <abbr title="convertit la chaine de caractère d'une requête HTTP en donnée Python">"décodage" JSON</abbr> plus rapide.
```

> * <a href="https://github.com/JoshData/python-email-validator" target="_blank"><code>email_validator</code></a> - por la validation par e-mail.

Reviewer 2: ```suggestion
* <a href="https://github.com/JoshData/python-email-validator" target="_blank"><code>email_validator</code></a> - pour la validation d'e-mail.
```

Utilisé par Starlette :

* <a href="https://requests.readthedocs.io" target="_blank"><code>requests</code></a> - Obligatoire si vous souhaitez utiliser le `TestClient`.
* <a href="https://github.com/Tinche/aiofiles" target="_blank"><code>aiofiles</code></a> - Obligatoire si vous souhaitez utiliser `FileResponse` ou `StaticFiles`.
* <a href="https://jinja.palletsprojects.com" target="_blank"><code>jinja2</code></a> - Obligatoire si vous souhaitez utiliser la configuration de modèle par défaut.

> * <a href="https://andrew-d.github.io/python-multipart/" target="_blank"><code>python-multipart</code></a> - Obligatoire si vous souhaitez soutenir le formulaire <abbr title="converting the string that comes from an HTTP request into Python data">"parsing"</abbr>, avec `request.form()`.

Reviewer 2: ```suggestion
* <a href="https://andrew-d.github.io/python-multipart/" target="_blank"><code>python-multipart</code></a> - Obligatoire si vous souhaitez supporter le <abbr title="convertit la chaine de caractère d'une requête HTTP en donnée Python">"décodage"</abbr> de formulaire, avec `request.form()`.
```

> * <a href="https://pythonhosted.org/itsdangerous/" target="_blank"><code>itsdangerous</code></a> - Requis pour la prise en charge de « SessionMiddleware ».

Reviewer 2: ```suggestion
* <a href="https://pythonhosted.org/itsdangerous/" target="_blank"><code>itsdangerous</code></a> - Requis pour la prise en charge de `SessionMiddleware`.
```

        ... "item_price": item.price ...
```

...et voyez comment votre éditeur complétera automatiquement les attributs et connaîtra leurs types :

![editor support](https://fastapi.tiangolo.com/img/vscode-completion.png)
Pour un exemple plus complet incluant plus de fonctionnalités, voir le <a href="https://fastapi.tiangolo.com/tutorial/">Tutoriel - User Guide</a>.

**Spoiler alert**: le tutoriel - user guide inclu:

* Déclaration de **paramètres** provenant d'autres endroits différents comme : **en-têtes**, **cookies**, **champs de formulaire** et **fichiers**.
* Comment définir les **contraintes de validation** comme `maximum_length` ou `regex`.
* Un très puissant et facile à utiliser **<abbr title="also known as components, resources, providers, services, injectables">Systéme d'injection de dépendance </abbr>** .
* Sécurité et authentification, y compris la prise en charge de **OAuth2** avec **jetons JWT** et l'authentification **HTTP Basic**.
* Des techniques plus avancées (mais tout aussi simples) pour déclarer des **modèles JSON profondément imbriqués** (grâce à Pydantic).

> * De nombreuses fonctionnalités supplémentaires (grâce à Starlette) comme:

Reviewer 3: ```suggestion
* d'obtenir beaucoup de fonctionnalités supplémentaires (grâce à Starlette) comme:
```

    * **WebSockets**
    * **GraphQL**

>     * tests extrêmement faciles basés sur `requests` et `pytest`

Reviewer 3: ```suggestion
    * de tester très facilement avec `requests` et `pytest`
```

    * **CORS**
    * **Cookie Sessions**
    * ...et plus encore.

## Performance

> Les benchmarks TechEmpower indépendants montrent les applications **FastAPI** s'exécutant sous Uvicorn comme <a href="https://www.techempower.com/benchmarks/#section=test&runid=7464e520-0dc2-473d-bd34-dbdfd7e85911&hw=ph&test=query&l=zijzen-7" class="external-link" target="_blank"> l'un des frameworks Python les plus rapides disponibles </a>, uniquement en dessous de Starlette et Uvicorn eux-mêmes (utilisés en interne par FastAPI). (*)

Reviewer 3: ```suggestion
Les benchmarks TechEmpower indépendants montrent que les applications **FastAPI** s'exécutant sous Uvicorn sont <a href="https://www.techempower.com/benchmarks/#section=test&runid=7464e520-0dc2-473d-bd34-dbdfd7e85911&hw=ph&test=query&l=zijzen-7" class="external-link" target="_blank"> parmi les frameworks existants en Python les plus rapides </a>, juste derrière Starlette et Uvicorn (utilisés en interne par FastAPI). (*)
```

Pour en savoir plus, consultez la section <a href="https://fastapi.tiangolo.com/benchmarks/" class="internal-link" target="_blank">Benchmarks</a>.

## Dépendances facultatives

> Utilisé par Pydantic:

Reviewer 3: ```suggestion
Utilisées par Pydantic:
```

* <a href="https://github.com/esnme/ultrajson" target="_blank"><code>ujson</code></a> - pour un <abbr title="convertit la chaine de caractère d'une requête HTTP en donnée Python">"décodage" JSON</abbr> plus rapide.

> * <a href="https://github.com/JoshData/python-email-validator" target="_blank"><code>email_validator</code></a> - pour la validation d'e-mail.

Reviewer 3: ```suggestion
* <a href="https://github.com/JoshData/python-email-validator" target="_blank"><code>email_validator</code></a> - pour la validation des adresses email.
```

> Utilisé par Starlette :

Reviewer 3: ```suggestion
Utilisées par Starlette :
```

* <a href="https://jinja.palletsprojects.com" target="_blank"><code>jinja2</code></a> - Obligatoire si vous souhaitez utiliser la configuration de modèle par défaut.
* <a href="https://andrew-d.github.io/python-multipart/" target="_blank"><code>python-multipart</code></a> - Obligatoire si vous souhaitez supporter le <abbr title="convertit la chaine de caractère d'une requête HTTP en donnée Python">"décodage"</abbr> de formulaire, avec `request.form()`.
* <a href="https://pythonhosted.org/itsdangerous/" target="_blank"><code>itsdangerous</code></a> - Requis pour la prise en charge de `SessionMiddleware`.
* <a href="https://pyyaml.org/wiki/PyYAMLDocumentation" target="_blank"><code>pyyaml</code></a> - Requis pour le support `SchemaGenerator` de Starlette (vous n'en avez probablement pas besoin avec FastAPI).
* <a href="https://graphene-python.org/" target="_blank"><code>graphene</code></a> - Requis pour la prise en charge de `GraphQLApp`.
* <a href="https://github.com/esnme/ultrajson" target="_blank"><code>ujson</code></a> - Obligatoire si vous souhaitez utiliser `UJSONResponse`.

> Utilisé par FastAPI / Starlette :

Reviewer 3: ```suggestion
Utilisées par FastAPI / Starlette :
```
